deploy/backup-s3.yaml

```diff
@@ -2,6 +2,8 @@ apiVersion: v1
 kind: Secret
 metadata:
   name: cluster1-s3-credentials
+  finalizers:
```

Comment:
We should:
1. add a check in a test that this finalizer is set
2. add a check that when the backup object is deleted from the kubernetes that the backup is actually deleted from the storage (I think we should do it for at least one storage type if not for all)